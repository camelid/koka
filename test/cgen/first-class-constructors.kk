// Tests C code generation of constructors treated as first-class values.

fun cons-alias() : ((a, list<a>) -> list<a>) {
    return Cons
}

fun main() {
    val list_of_maybes = map(cons-alias()(1, [2, 3]), Just)
    val list_of_ints = map(
        list_of_maybes,
        fn(maybe_el) { match(maybe_el) { Just(el) -> el } }
    )
    println(show(list_of_ints))
}
