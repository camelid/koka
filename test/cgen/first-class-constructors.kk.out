[1,2,3]
add default effect for std/core/exn